src/access_nri_intake/data/utils.py

```diff
@@ -48,21 +48,41 @@ def available_versions(pretty: bool = True):
     # Work out where the catalogs are stored
     base_path = _get_catalog_rp()
 
+    # Grab the extant catalog and work out its min and max versions
+    try:
+        with open(get_catalog_fp()) as cat:
+            cat_yaml = yaml.safe_load(cat)
+            vers_min = cat_yaml["sources"]["access_nri"]["parameters"]["version"]["min"]
+            vers_max = cat_yaml["sources"]["access_nri"]["parameters"]["version"]["max"]
+            vers_def = cat_yaml["sources"]["access_nri"]["parameters"]["version"][
+                "default"
+            ]
+    except FileNotFoundError:
+        raise FileNotFoundError(f"Unable to find catalog at {get_catalog_fp()}")
+    except KeyError:
+        raise RuntimeError(f"Catalog at {get_catalog_fp()} not correctly formatted")
+
     # Grab all the catalog names
-    cats = [d for d in os.listdir(base_path) if re.search(CATALOG_NAME_FORMAT, d)]
+    cats = [
+        d.name
```

Comment:
Minor nitpick but I find `for dir in ..` rather than `for d in ...` more readable - the list comprehension spans a few lines already so no harm in readability from the extra characters.

Reply:
NB. I only find this holds true with multi line expressions - maybe I need to get my eyes checked.

Reply:
Fair comment, although is `dir` a dangerous/possibly protected name? Something like `dir_path` is probably safer (and makes it clearer that `d`/`dir`/`dir_path` is a `Path` object).

Reply:
I *think* that dir is a soft keyword, a la match, but yeah, that is a good point - `dir_path` seems sensible.